projects/community-payback-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/AppointmentsService.kt

```diff
@@ -119,61 +118,55 @@ class AppointmentsService(
-            ?: throw NotFoundException("UPWAppointment", "appointmentId", appointmentId)
+        val appointment = unpaidWorkAppointmentRepository.getAppointment(appointmentId)
 
         val contact = appointment.contact
 
-        if (!appointmentOutcome.endTime.isAfter(appointmentOutcome.startTime)) {
-            throw IllegalStateException("End Time must be after Start Time")
+        require(request.endTime > request.startTime) {
+            "End Time must be after Start Time"
         }
 
-        if (LocalDateTime.of(appointment.appointmentDate, appointmentOutcome.endTime).isBefore(LocalDateTime.now())
-            && appointmentOutcome.outcome == null
+        require(
+            request.outcome != null ||
+                LocalDateTime.of(appointment.appointmentDate, request.endTime) > LocalDateTime.now()
         ) {
-            throw IllegalStateException("Appointments in the past require an outcome")
+            "Appointments in the past require an outcome"
         }
 
-        val outcome = appointmentOutcome.outcome?.let {
-            contactOutcomeRepository.findContactOutcomeByCode(it.code)
-                ?: throw IllegalStateException("Contact outcome with code ${it.code} not found")
+        val outcome = request.outcome?.let {
+            contactOutcomeRepository.getContactOutcome(it.code)
         }
 
-        val workQualityCode = WorkQuality.of(appointmentOutcome.workQuality)?.name
-        val behaviourCode = Behaviour.of(appointmentOutcome.behaviour)?.name
-
-        val workQuality = workQualityCode?.let {
-            referenceDataRepository.findByCodeAndDatasetCode(workQualityCode, Dataset.UPW_WORK_QUALITY)
-                ?: throw IllegalStateException("Work quality reference data with code $workQualityCode not found")
-        } ?: throw IllegalStateException("Work quality ${appointmentOutcome.workQuality} not recognised")
+        val workQuality = request.workQuality?.let {
+            referenceDataRepository.getWorkQuality(request.workQuality.code)
+        }
 
-        val behaviour = behaviourCode?.let {
-            referenceDataRepository.findByCodeAndDatasetCode(behaviourCode, Dataset.UPW_BEHAVIOUR)
-                ?: throw IllegalStateException("Behaviour reference data with code $behaviourCode not found")
-        } ?: throw IllegalStateException("Behaviour ${appointmentOutcome.behaviour} not recognised")
+        val behaviour = request.behaviour?.let {
+            referenceDataRepository.getBehaviour(request.behaviour.code)
+        }
 
         val staff =
-            appointmentOutcome.supervisor?.let {
-                staffRepository.findStaffByCode(appointmentOutcome.supervisor.code)
-                    ?: throw IllegalStateException("Staff with code ${appointmentOutcome.supervisor.code} not found")
+            request.supervisor.let {
+                staffRepository.getStaff(request.supervisor.code)
             }
```

Comment:
You don't need the let here, now that it's non-nullable
```suggestion
        val staff = staffRepository.getStaff(request.supervisor.code)
```